forge-game/src/main/java/forge/game/ability/effects/ChooseTypeEffect.java

```diff
@@ -154,4 +154,10 @@ public void resolve(SpellAbility sa) {
             throw new InvalidParameterException(sa.getHostCard() + "'s ability resulted in no types to choose from");
         }
     }
+
+    private static List<String> getInvalidTypes(SpellAbility sa) {
+        return sa.hasParam("InvalidTypes") ?
+                Arrays.asList(sa.getParam("InvalidTypes").split(",")) :
+                new ArrayList<>();
+    }
```

Comment:
Would it be better to use Collections.emptyList() if it isn't mutable?

Reply:
This snippet was already like that, I just moved it to a function. If you think it's worth the change, I'll investigate